context/fromIpfs.ts

```diff
@@ -0,0 +1,64 @@
+import { MichelsonMap } from "@taquito/taquito";
+import { buf2hex } from "@taquito/utils";
+import { IPFS } from "./config";
+
+// Check if running in a Node.js environment
+const isNode =
+  typeof process !== "undefined" &&
+  process.versions != null &&
+  process.versions.node != null;
+
+let FormDataNode: new () => any;
+let fetch:
+  | (((
+      input: RequestInfo | URL,
+      init?: RequestInit | undefined
+    ) => Promise<Response>) &
+      ((
+        input: RequestInfo | URL,
+        init?: RequestInit | undefined
+      ) => Promise<Response>))
+  | ((arg0: string, arg1: { method: string; body: any; headers: {} }) => any);
+
+if (isNode) {
+  FormDataNode = require("form-data");
+  fetch = require("node-fetch");
+} else {
+  fetch = window.fetch; // Use the browser's fetch
+}
```

Comment:
Why do we need to have a specific case for NodeJs? it supposed to run only in the browser

Reply:
To be able to run e2e test on the docker image. Otherwise, CI keeps complaining and fail.
see the build, like
- https://github.com/marigold-dev/tzsafe-ui/actions/runs/7518831530/job/20466630403

in CI, it will keep complaining about `fromdata`, `blob` and `fetch`.

Reply:
This should be done a test time with mocking as there's no reason to put it at runtime, but it can be fixed later